src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -1,455 +1,702 @@
-				//we have a socket and a 200 status. 
-				//Could still be a false positive though, if the server ignored the method, 
-				//and did not recognise the URL, so redirected to a login page, for instance
-				//Remediation: Check the contents match the expected third party contents.
-				if (log.isDebugEnabled()) log.debug("Raw Socket established, in theory to "+thirdpartyHost);
-				
-				OutputStream os = socket.getOutputStream();
-				InputStream is = socket.getInputStream();
-				
-				PrintWriter pw = new PrintWriter(os, false);
-				pw.write("GET http://"+thirdpartyHost + ":"+thirdpartyPort + "/ HTTP/1.1\n");
-				pw.write("Host: "+thirdpartyHost+"\n\n");
-				pw.flush();
-				
-				//read the response via a 4k buffer
-				ByteArrayOutputStream bos = new ByteArrayOutputStream ();
-				byte [] buffer = new byte [1024*4];
-				int bytesRead = is.read(buffer);
-				int totalBytesRead = 0;
-				while (bytesRead > -1) {
-					totalBytesRead+= bytesRead;					
-					bos.write(buffer, 0, bytesRead);					
-					bytesRead = is.read(buffer);
-				}
-				String response = new String (bos.toByteArray());				
-				if (log.isDebugEnabled()) log.debug("Response is "+totalBytesRead+" bytes: \n"+response);
-				
-				Matcher m = thirdPartyContentPattern.matcher(response);
-				if (m.matches()) {
-					if (log.isDebugEnabled()) log.debug("Response matches expected third party pattern!");
-					is.close();
-					os.close();
-					bos.close();
-					socket.close();
-					return true;
-				} else { 
-					if (log.isDebugEnabled()) log.debug("Response does *not* match expected third party pattern");
-				}
-				is.close();
-				os.close();
-				bos.close();
-				socket.close();
-				return false;				
-			}
-			else  {
-				if (log.isDebugEnabled()) { 
-					log.debug ("Could not establish a socket connection to a third party using the CONNECT HTTP method: NULL socket returned, or non-200 response");
-					log.debug ("The status line returned: "+statusLine);
-				}
-				return false;
-			} 
-		}
-		catch (Exception e) {
-			if (log.isDebugEnabled()) log.debug ("Could not establish a socket connection to a third party using the CONNECT HTTP method", e);
-		}
-		return false;
-	}
-
-}
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.ascanbeta;

```

Comment:
The package needs to be `ascanrulesBeta` (instead of `ascanbeta`).

Reply:
Yes my mistake, I'll fix it.